python/hail/expr/functions.py

```diff
@@ -633,14 +633,13 @@ def index(structs, identifier):
         raise TypeError("'index' expects an array with element type 'Struct', found '{}'"
                         .format(structs.dtype))
     struct_type = structs._type.element_type
-    struct_fields = {fd.name: fd.dtype for fd in struct_type.fields}
```

Comment:
we can just get rid of `index`. Is it used anywhere?

Reply:
Ah, using the dict constructor.  niiice.

Reply:
I kinda remember it being used in the view joins. That might be a pain to try to do here.

Reply:
Yeah, I just discovered that.  I'm going to leave it in for now.